packages/kotti-ui/source/kotti-field-inline-edit/KtFieldInlineEdit.vue

```diff
@@ -0,0 +1,360 @@
+					return fieldValidation === 'error'
+				case KottiFieldInlineEdit.ConfirmationValidation.WARNING:
+					return ['error', 'warning'].includes(fieldValidation)
+				case KottiFieldInlineEdit.ConfirmationValidation.NEVER:
+					return false
+				default:
+					throw new Error(
+						`KtFieldInlineEdit: Unexpected type for 'preventConfirmationOn': ${props.preventConfirmationOn}`,
+					)
+			}
+		})
+
+		const setFieldValue = (newValue: KottiFieldInlineEdit.Value) => {
+			field.setValue(newValue)
+			forceUpdate()
+		}
+
+		const updateIsEditing = (
+			shouldEdit: KottiFieldInlineEdit.Events.UpdateIsEditing,
+		) => {
+			if (shouldEdit === props.isEditing) {
+				/*
+					condition shouldn't be moved outside of the wrapping if.
+					In case of not emitting the update:isEditing, the component
+					has to re-focus on the input field if needed.
+
+					Otherwise, the component emits and if the isEditing is set by
+					the user app, the same condition will be applied in the watcher.
+				*/
+				if (shouldEdit && !isInFocus(inputRef.value)) inputRef.value?.focus()
+
+				return
+			}
+
+			emit('update:isEditing', shouldEdit)
+		}
+
+		const preEditingValue = ref<KottiFieldInlineEdit.Value>(field.currentValue)
+
+		watch(
+			() => props.isEditing,
+			(shouldEdit, wasEditing) => {
+				if (shouldEdit && !isInFocus(inputRef.value)) inputRef.value?.focus()
+
+				if (shouldEdit === wasEditing) return
+
+				preEditingValue.value = field.currentValue
+			},
+			{ immediate: true },
+		)
+
+		const onConfirm = () => {
+			if (preventConfirm.value) {
+				if (!isInFocus(inputRef.value)) inputRef.value?.focus()
+				return
+			}
+
+			emit('confirm', field.currentValue)
+
+			updateIsEditing(false)
+
+			blurField(fieldRef.value)
+		}
+
+		const onCancel = () => {
+			setFieldValue(preEditingValue.value)
+
+			updateIsEditing(false)
+
+			blurField(fieldRef.value)
+		}
+
+		const onClick = (event: MouseEvent | KeyboardEvent) => {
+			if (event.target === null || props.isDisabled) return
+
+			const isClickOutside = !isEventTarget(fieldRef.value, event.target)
+			lastEventTarget.value = event.target
+
+			if (isClickOutside) onCancel()
+			else updateIsEditing(true)
+		}
+
+		/**
+		 * The last DOM element to capture focus/click events
+		 */
+		const lastEventTarget = ref<EventTarget | null>(null)
+
+		/**
+		 * `tab` key presses don't trigger on mobile,
+		 * and @blur events are tricky to implement given the implementation of
+		 * `KtField`. Therefore, a listener is attached to DOM `focus` events.
+		 *
+		 * The target element is continuously maintained in a variable to compare
+		 * changes and distinguish between ignore and confirm cases.
+		 *
+		 * In case of clicking on a focusable element, the focus event is captured first then the click event.
+		 *
+		 * In case of clicking on a non-focusable element, the focus event doesn't trigger.
+		 */
+		const onFocusChange = (event: Event) => {
```

Comment:
Can we move
```
/**
		 * The last DOM element to capture focus/click events
		 */
		const lastEventTarget = ref<EventTarget | null>(null)
```
above both onClick and onFocusChange bc it's used in both. 

Also maybe moving the description above onFocusChange above both, the description text is related to both click and focus events.